src/accessibility/describe.js

```diff
@@ -0,0 +1,304 @@
+    document.getElementById(cnvId).innerHTML =
+      '<div id="' +
+      cnvId +
+      ds +
+      '" role="region" aria-label="Canvas Description"><table id="' +
+      cnvId +
+      ft +
+      '"><caption>Canvas elements and their descriptions</caption></table></div>';
+  } else if (document.getElementById(cnvId + ft) === null) {
+    document
+      .getElementById(cnvId + fds)
+      .insertAdjacentHTML(
+        'afterend',
+        '<table id="' +
+          cnvId +
+          ft +
+          '"><caption>Canvas elements and their descriptions</caption></table>'
+      );
+  }
+  if (document.getElementById(cnvId + fte + n) === null) {
+    let tr = document.createElement('tr');
+    tr.id = cnvId + fte + n;
+    document.getElementById(cnvId + ft).appendChild(tr);
+  }
+  if (
+    document.getElementById(cnvId + fte + n).innerHTML !==
+    '<th scope="row">' + nm + '</th><td>' + t + '</td>'
+  ) {
+    document.getElementById(cnvId + fte + n).innerHTML =
+      '<th scope="row">' + nm + '</th><td>' + t + '</td>';
+  }
+  //If display is LABEL creates a div adjacent to the canvas element with
+  //a table, a row header cell with the name of the elements,
+  //and adds the description of the element in adjecent cell.
+  if (d === this.LABEL) {
+    if (document.getElementById(cnvId + lb) === null) {
+      document
+        .getElementById(cnvId)
+        .insertAdjacentHTML(
+          'afterend',
+          '<div id="' +
+            cnvId +
+            lb +
+            '" class="p5Label"><table id="' +
+            cnvId +
+            lt +
+            '"></table></div>'
+        );
+    } else if (document.getElementById(cnvId + lt) === null) {
+      document
+        .getElementById(cnvId + lds)
+        .insertAdjacentHTML(
+          'afterend',
+          '<table id="' + cnvId + lt + '"></table>'
+        );
+    }
+    if (document.getElementById(cnvId + lte + n) === null) {
+      let tr = document.createElement('tr');
+      tr.id = cnvId + lte + n;
+      document.getElementById(cnvId + lt).appendChild(tr);
+    }
+
+    if (
+      document.getElementById(cnvId + lte + n).innerHTML !==
+      '<th scope="row">' + nm + '</th><td>' + t + '</td>'
+    ) {
+      document.getElementById(cnvId + lte + n).innerHTML =
+        '<th scope="row">' + nm + '</th><td>' + t + '</td>';
+    }
+  }
+};
+/**
+ * Helper function for describeElement().
+ */
+p5.prototype._elementName = function(n) {
+  if (n === this.LABEL || n === this.FALLBACK) {
+    throw new Error('element name should not be LABEL or FALLBACK');
+  }
+  let lm = n[n.length - 1];
+  //check if last character of string n is '.', ';', or ','
+  if (lm === '.' || lm === ';' || lm === ',') {
+    //replace last character with ':'
+    n = n.replace(/.$/, ':');
+    //if string n does not end with ':'
+  } else if (n.endsWith(':') === false) {
+    //add ':'' at the end of string
+    n = n + ':';
+  }
+  //if first character of string is not capitalized
```

Comment:
does capitalization matter if this is for screen reader? pardon my lack of knowledge about this

Reply:
it doesn't really matter for something like this. after thinking about it I've decided to leave it up to the user and deleted these regex lines.